File: j2ee-spring-patterns/business-delegate/go/delegate/MessageDelegate.go
package delegate

import (
	"business-delegate/model"
	"business-delegate/service"
)

type MessageDelegate struct {
	service       service.MessageService
	lookupService service.MessageLookupService
}

func (md MessageDelegate) SendMessage(messageType model.MessageType) {
	md.service = md.lookupService.GetMessageService(messageType)
	md.service.SendMessage()
}

File: j2ee-spring-patterns/business-delegate/python/delegate/message_delegate.py
from service.message_service_lookup import MessageServiceLookup
from service.message_service import MessageService


class MessageDelegate:
    def __init__(self, message_type):
        self.message_service_lookup = MessageServiceLookup
        self.message_service = MessageService
        self.message_type = message_type

    def set_message_type(self, message_type):
        self.message_type = message_type

    def send_message(self):
        self.message_service = self.message_service_lookup.get_message_service(message_type=self.message_type)
        self.message_service.send_message()
